app/presenters/research_session_presenter.rb

```diff
@@ -26,4 +30,40 @@ def recording_methods_list
     end
     lowercase_words.to_sentence
   end
+
+  def any_expenses?
+    !!(
+      travel_expenses_limit&.nonzero? ||
+        food_expenses_limit&.nonzero? ||
+        other_expenses_limit&.nonzero?
+    )
+  end
+
+  def expenses_sentence
+    return nil unless any_expenses?
+    fragments = [
+      :travel_expenses,
+      :food_expenses,
+      :other_expenses
+    ].map do |expense_attr|
+      value = send("#{expense_attr}_limit")
+      if value&.nonzero?
```

Comment:
Always feels like stuff like this should happen in the view layer.

Reply:
Do you mean to do the `map` in the view layer too? I am impartial as I can't tell what it would look like in the view.

Reply:
Just feels like something that belongs in a view template

Reply:
As a `Presenter`, this is view layer. Outside of the `html.erb` files, you can either do pure Ruby view logic in `Helper` modules (as we have with [`edit_link_for`](https://github.com/barnardos/consent-form-builder-rails/blob/7819c06e7834ccd853f2f81b3691f7046a04c51a/app/helpers/research_sessions_helper.rb#L2), or – as in this case where the view logic relates to a specific instance of model class `ResearchSession` – by wrapping that `ResearchSession` in a `*Presenter` decorator.

I've been trying to find the best article about the pattern but they're all pretty old now...